repo/blob/azure/azure_storage_test.go

```diff
@@ -270,3 +272,175 @@ func TestAzureStorageInvalidCreds(t *testing.T) {
 		t.Errorf("unexpected success connecting to Azure blob storage, wanted error")
 	}
 }
+
+// TestAzureStorageRansomwareProtection runs through the behaviour of Azure ransomware protection.
+// 1. blob is created then the retention is extended.
+// 2. blob is logically deleted while the retention period is in place, by creating a delete marker (d_) file.
+// 3. delete marker blob is extended.
+// 4. original blob is deleted.
+// 5. delete marker blob further extension fails, then it is deleted.
+func TestAzureStorageRansomwareProtection(t *testing.T) {
+	t.Parallel()
+	testutil.ProviderTest(t)
+
+	// must be without locked policy or the retention period will be too high (1+ days)
+	container := getEnvOrSkip(t, testContainerEnv)
```

Comment:
I'm not sure where this is populated from, it may need a separate container if the current one used for tests does not have `ImmutableStorageWithVersioning` enabled or if it has a locked policy